versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractReferences.java

```diff
@@ -159,30 +158,58 @@ public void createAndDeleteTag() throws Exception {
+    assertThat(store().hashOnReference(branch, Optional.empty())).isEqualTo(hashFromCreate);
+
+    final Hash firstCommitHash = commit("First Commit").toBranch(branch);
+
+    assertThat(store().getNamedRef(branch.getName(), GetNamedRefsParams.DEFAULT))
+        .extracting(ReferenceInfo::getHash, ReferenceInfo::getNamedRef)
+        .containsExactly(firstCommitHash, branch);
+
+    final Hash secondCommitHash = commit("Second Commit").toBranch(branch);
+    final Hash thirdCommitHash = commit("Third Commit").toBranch(branch);
 
-    final Hash firstCommit = commit("First Commit").toBranch(branch);
+    BranchName branchName = BranchName.of("getNamedRef_branch_" + secondCommitHash.asString());
+    TagName tagName = TagName.of("getNamedRef_tag_" + thirdCommitHash.asString());
 
-    assertThat(store().toRef(branch.getName())).isEqualTo(WithHash.of(firstCommit, branch));
+    store().create(branchName, Optional.of(secondCommitHash));
+    store().create(tagName, Optional.of(thirdCommitHash));
 
-    final Hash secondCommit = commit("Second Commit").toBranch(branch);
-    final Hash thirdCommit = commit("Third Commit").toBranch(branch);
+    // Verifies that the result of "getNamedRef" for the branch created at "firstCommitHash" is
+    // correct
+    assertThat(store().getNamedRef(branchName.getName(), GetNamedRefsParams.DEFAULT))
+        .extracting(ReferenceInfo::getHash, ReferenceInfo::getNamedRef)
+        .containsExactly(secondCommitHash, branchName);
 
-    store().create(BranchName.of(thirdCommit.asString()), Optional.of(firstCommit));
-    store().create(TagName.of(secondCommit.asString()), Optional.of(firstCommit));
+    // Verifies that the result of "getNamedRef" for the tag created at "firstCommitHash" is correct
+    assertThat(store().getNamedRef(tagName.getName(), GetNamedRefsParams.DEFAULT))
+        .extracting(ReferenceInfo::getHash, ReferenceInfo::getNamedRef)
+        .containsExactly(thirdCommitHash, tagName);
 
-    assertThat(store().toRef(secondCommit.asString()))
-        .isEqualTo(WithHash.of(firstCommit, TagName.of(secondCommit.asString())));
-    assertThat(store().toRef(thirdCommit.asString()))
-        .isEqualTo(WithHash.of(firstCommit, BranchName.of(thirdCommit.asString())));
-    // Is it correct to allow a reference with the sentinel reference?
-    // assertThat(store().toRef(initialCommit.asString()), is(WithHash.of(initialCommit,
-    // initialCommit)));
-    assertThrows(ReferenceNotFoundException.class, () -> store().toRef("unknown-ref"));
-    assertThrows(ReferenceNotFoundException.class, () -> store().toRef("1234567890abcdef"));
+    // Verifies that the result of "getNamedRef" for the branch created at "firstCommitHash" is
+    // correct
+    assertThat(store().getNamedRef(branchName.getName(), GetNamedRefsParams.DEFAULT))
+        .extracting(ReferenceInfo::getHash, ReferenceInfo::getNamedRef)
+        .containsExactly(secondCommitHash, branchName);
```

Comment:
Is this not the same as lines 188-190? :thinking:

Reply:
Oops, copy-paste-keyboard-bug